### PR TITLE
Added functions to clear a bit, set a bit and to check status of bit

### PR DESCRIPTION
**Pull request by abhishekk07**
Added functions to clear a bit, set a bit and to check status of bit

### CODE REVIEW

**Comment by MaJerle** [comment]
Thanks for your contribution. A check-bit function could be implemented by macro, no?

**Comment by abhishekk07** [comment]
Function implementation is better because it returns only 1 or 0. 
If check bit is implemented as a macro, then the result of the bit wise operation will not always be 1(It can be 1,2,4,8.. etc), due to this when check_bit() is used in a comparison, the other operator also has to be variable/macro with the specific bit set.

I have used some issues previously while using macro based bit checking.

**Comment by MaJerle** [comment]
I tentatively agree with you. Simple implementation could be with preprocess, such as example below.

Let me know what do you think. Result of operations is always 1 or 0.

```
/* Mask result of bitwise AND must equal to mask => all bits are set */
#define lwutil_check_bit_mask_all(val, mask)         (((val) & (mask)) == (mask))

/* Mask result of bitwise AND must not be zero => at least one bit from mask is set */
#define lwutil_check_bit_mask_any(val, mask)         (((val) & (mask)) != 0)

/* And then, to make a bit */
#define lwutil_check_bit_pos(val, bit_pos)           lwutil_check_bit_mask_all((val), 1 << (bit_pos))
```

File: lwutil.h [hunk 1]
```diff
@@ -88,6 +88,22 @@ extern "C" {
...
 #define LWUTIL_CONCAT_BASE(x, y)                x ## y
 #define LWUTIL_CONCAT(s0, s1)                   LWC_CONCAT_BASE(s0, s1)
+/**
+ * \brief           Set a bit at specific position in a number, other bits are unchanged
+ * \param[in]       num: Number in which bit needs to be set
+ * \param[in]       pos: Position of bit to be set
+ * \return          number with bit at 'pos' set
+ */
+
+#define LWUTIL_SET_BIT(num, pos)				num = num | (1<<pos)
```
Comment (MaJerle) [inline lwutil.h:98]: This code is not safe, especially if num is an expression.

File: lwutil.h [hunk 2]
```diff
@@ -88,6 +88,22 @@ extern "C" {
...
+/**
+ * \brief           Clear a bit at specific position in a number, other bits are unchanged
+ * \param[in]       num: Number in which bit needs to be cleared
+ * \param[in]       pos: Position of bit to be cleared
+ * \return          number with bit at 'pos' cleared
+ */
+
+#define LWUTIL_CLR_BIT(num,pos)					num = num & ~(1<<pos)
```
Comment (MaJerle) [inline lwutil.h:106]: This code is not safe, especially if num is an expression.

File: lwutil.h (hunk 1 above)
Comment (abhishekk07) [reply lwutil.h:98]: Can you elaborate? I am not sure in which scenario num will be an expression? 
As i understand it we can only set/clear bits on variables

**Comment by abhishekk07** [comment]
> 
> 
> I tentatively agree with you. Simple implementation could be with preprocess, such as example below.
> 
> Let me know what do you think. Result of operations is always 1 or 0.
> 
> ```
> /* Mask result of bitwise AND must equal to mask => all bits are set */
> #define lwutil_check_bit_mask_all(val, mask)         (((val) & (mask)) == (mask))
> 
> /* Mask result of bitwise AND must not be zero => at least one bit from mask is set */
> #define lwutil_check_bit_mask_any(val, mask)         (((val) & (mask)) != 0)
> 
> /* And then, to make a bit */
> #define lwutil_check_bit_pos(val, bit_pos)           lwutil_check_bit_mask_all((val), 1 << (bit_pos))
> ```

Looks good and i agree that this is a better implementation through macro.

**Pull request closed without merging**